### ISSUE TITLE
fix(solvers): `pv_inversion` with a traced scalar `lambda_` fails (`float(lam)` at elliptic.py:760/783/799)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#309]
Part of #277.

## Summary

`pv_inversion` (`finitevolx/_src/solvers/elliptic.py:685-916`) calls `float(lam)` in three places (`:760, :783, :799`) when normalising a scalar `lambda_`. Under `jax.jit` or `vmap` with a traced scalar this raises `ConcretizationTypeError`; a traced **array** λ works, and every other wrapper accepts a traced scalar.

## Reproduction

```python
import jax, jax.numpy as jnp
from finitevolx import pv_inversion
rhs = jnp.zeros((2, 34, 34)); dx = dy = 1.0
jax.jit(lambda r, l: pv_inversion(r, dx, dy, lambda_=l))(rhs, 4.0)   # ConcretizationTypeError
```

## Proposed fix

Drop the `float()` calls; use `jnp.asarray(lam)` and `jnp.ndim` checks (static) for the scalar/array/per-layer dispatch. While there, the triplicated layer loop (`:747-916`, 232 lines) is a candidate for one `vmap` over layers (see #280).

## Acceptance criteria

- [ ] `jax.jit` and `jax.vmap` over a scalar `lambda_` work (test).
- [ ] Scalar, `[nl]` array and per-layer callable paths give identical results to today.
